src/cli/telemetry/helpers.ts

```diff
@@ -0,0 +1,52 @@
+import fs from "fs";
+
+interface TerminalInfo {
+	/**
+	 * Whether this is in CI or not.
+	 */
+	readonly ci: boolean;
+	/**
+	 * Path to the user's shell program.
+	 */
+	readonly shell: string;
+	/**
+	 * Whether the terminal is an interactive TTY or not.
+	 */
+	readonly tty: boolean;
+	/**
+	 * Whether this is a Windows shell or not.
+	 */
+	readonly windows: boolean;
+}
+
+declare const TERMINAL_INFO: TerminalInfo;
+
+export const tryFn = async <T extends (...args: any[]) => Promise<R>, R>(
+	fn: T,
+	...args: any[]
+): Promise<R | null> => {
+	try {
+		return await fn(...args);
+	} catch {
+		// ignore
+	}
+
+	return null;
+};
+
+export const isInteractive = (): boolean => TERMINAL_INFO.tty && !TERMINAL_INFO.ci;
+
+// Plucked from https://github.com/ionic-team/capacitor/blob/b893a57aaaf3a16e13db9c33037a12f1a5ac92e0/cli/src/util/uuid.ts
+export function uuidv4(): string {
+	return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, c => {
+		const r = (Math.random() * 16) | 0;
+		const v = c == 'x' ? r : (r & 0x3) | 0x8;
+
+		return v.toString(16);
+	});
+}
+
+export async function readJson(path: string) {
+	const rawdata = await fs.promises.readFileSync(path);
```

Comment:
CI is [currently failing](https://github.com/ionic-team/stencil/pull/2963/checks?check_run_id=3080055348)

> updateConfig › should edit a file
> 
>    TypeError: fs_1.default.promises.readFileSync is not a function


Did you mean:

```suggestion
	const rawdata = await fs.promises.readFile(path);
```

https://nodejs.org/api/fs.html#fs_fspromises_readfile_path_options

Reply:
NVM, looks like this got fixed while I was typing it 😆